### DISCUSSION TITLE
Why not set all props from constructor ?

### DISCUSSION THREAD

**Discussion by dridk** [zauberzeug/nicegui#1910] [Ideas / Feature Requests]
Hi, 

I wonder why not setting all quasar props from the constructor? Some props from quasar cannot be set from the python constructor.  I have to use props() method for that. For instance the flat props from Qbtn.

**Comment by rodja** [comment]
It's mostly a workload restriction. Adding all the props will be quite time consuming. But there are other things to consider: not all props make sense to be used with NiceGUI, and the number of parameters will get quite high. So instead of putting every option into the constructor there could also be properties to access and alter props. Maybe similar to how `.tailwind()` works:

```py
tap = ui.button('tap me', on_click=lambda: ui.notify("bam"))
tap.props.style_flat().content_no_caps().state_disable()
```

**Comment by dridk** [comment]
Ok it makes sense ! Thank you
